lib/ocpp/v201/charge_point.cpp

```diff
@@ -3207,6 +3232,97 @@ void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest
     this->send<SetChargingProfileResponse>(call_result);
 }
 
+void ChargePoint::handle_clear_charging_profile_req(Call<ClearChargingProfileRequest> call) {
+    EVLOG_debug << "Received ClearChargingProfileRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
+    const auto msg = call.msg;
+    ClearChargingProfileResponse response;
+    response.status = ClearChargingProfileStatusEnum::Unknown;
+
+    // K10.FR.06
+    if (msg.chargingProfileCriteria.has_value() and
+        msg.chargingProfileCriteria.value().chargingProfilePurpose.has_value() and
+        msg.chargingProfileCriteria.value().chargingProfilePurpose.value() ==
+            ChargingProfilePurposeEnum::ChargingStationExternalConstraints) {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = "InvalidValue";
+        response.statusInfo->additionalInfo = "ChargingStationExternalConstraintsInClearChargingProfileRequest";
+        EVLOG_debug << "Rejecting SetChargingProfileRequest:\n reasonCode: " << response.statusInfo->reasonCode.get()
+                    << "\nadditionalInfo: " << response.statusInfo->additionalInfo->get();
+    } else {
+        response = this->smart_charging_handler->clear_profiles(msg);
+    }
+
+    ocpp::CallResult<ClearChargingProfileResponse> call_result(response, call.uniqueId);
+    this->send<ClearChargingProfileResponse>(call_result);
+}
+
+void ChargePoint::handle_get_charging_profiles_req(Call<GetChargingProfilesRequest> call) {
+    EVLOG_debug << "Received GetChargingProfilesRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
+    const auto msg = call.msg;
+    GetChargingProfilesResponse response;
+
+    const auto profiles_to_report = this->smart_charging_handler->get_reported_profiles(msg);
+
+    if (not profiles_to_report.empty()) {
```

Comment:
This is a bit confusing for me. I would only to the exit path in the if and then the normal flow outside of the if/else.